tcping.go

```diff
@@ -387,10 +428,19 @@ func tcping(tcpStats *stats) {
 		/* if the previous probe was successful
 		and the current one failed: */
 		if !tcpStats.wasDown {
+			/* Update startOfDowntime */
 			tcpStats.startOfDowntime = getSystemTime()
+
+			/* Calculate the longest uptime */
+			tcpStats.endOfUptime = getSystemTime()
+			calcLongestUptime(tcpStats)
+
 			tcpStats.wasDown = true
 		}
 
+		tcpStats.endOfDowntime = getSystemTime()
```

Comment:
I don't think this should be called here. After all, we reach this code when the probe has **failed** or is failing. Line 463 is the correct place that longest downtime should be called.

Reply:
This assumes that the failure probe is continued.
I think the longestDowntime needs to be updated even if the probe does not succeed afterwards, because printStatistics needs that value.